.github/workflows/pre-generate-preview-link.yml

```diff
@@ -6,24 +6,29 @@ permissions:
 on:
     pull_request:
         types: [opened, synchronize]
-
+env:
+    HEAD_REF: ${{ github.head_ref }}
 concurrency:
-    group: cloudflare-pages-verify-${{ github.head_ref }}
+    group: cloudflare-pages-verify-"$HEAD_REF"
     cancel-in-progress: true
 
 jobs:
     verify_pull_request:
         runs-on: Ubuntu-latest
         steps:
             - name: Retrieve PR information
+              env:
+                  EVENT_NUMBER: ${{ github.event.number }}
+                  EVENT_USERNAME: ${{ github.event.pull_request.user.login }}
               run: |
                   mkdir -p ./pr
-                  echo ${{ github.event.number }} > ./pr/NR
-                  echo ${{ github.event.pull_request.user.login }} > ./pr/USERNAME
-                  echo ${{ github.head_ref }} > ./pr/BRANCHNAME
```

Comment:
We still need branch name for smoke test to work fine. It's missing in the new lines.

Reply:
my miss, fixed.